collections/docs/03-keyset.md

```diff
@@ -0,0 +1,82 @@
+# KeySet
+
+The second type of collection is `collections.KeySet`, as the word suggests it maintains
+only a set of keys without values.
+
+#### Implementation curiosity
+
+A `collections.KeySet` is just a `collections.Map` with a `key` but no value.
+The value internally is always the same and is represented as an empty byte slice ```[]byte{}```.
+
+## Example
+
+As always we explore the collection type through an example:
+
+```go
+package collections
+
+import (
+	"cosmossdk.io/collections"
+	storetypes "cosmossdk.io/store/types"
+	"fmt"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+var ValidatorsSetPrefix = collections.NewPrefix(0)
+
+type Keeper struct {
+	Schema        collections.Schema
+	ValidatorsSet collections.KeySet[sdk.ValAddress]
+}
+
+func NewKeeper(storeKey *storetypes.KVStoreKey) Keeper {
+	sb := collections.NewSchemaBuilder(sdk.OpenKVStore(storeKey))
+	return Keeper{
+		ValidatorsSet: collections.NewKeySet(sb, ValidatorsSetPrefix, "validators_set", sdk.ValAddressKey),
+	}
+}
+
+func (k Keeper) AddValidator(ctx sdk.Context, validator sdk.ValAddress) error {
+	has, err := k.ValidatorsSet.Has(ctx, validator)
+	if err != nil {
+		return err
+	}
+	if has {
```

Comment:
is there a reason we wouldnt use an `else if` here and below?

Reply:
the first `if` at line 41 is returning, generally speaking in golang `else if` patterns are not common.